airflow/providers/microsoft/azure/hooks/azure_fileshare.py

```diff
@@ -96,6 +96,26 @@ def list_directories_and_files(self, share_name, directory_name=None, **kwargs):
         """
         return self.get_conn().list_directories_and_files(share_name, directory_name, **kwargs)
 
+    def list_files(self, share_name, directory_name=None, **kwargs):
+        """
+        Return the list of files stored on a Azure File Share.
+
+        :param share_name: Name of the share.
+        :type share_name: str
+        :param directory_name: Name of the directory.
+        :type directory_name: str
+        :param kwargs: Optional keyword arguments that
+            `FileService.list_directories_and_files()` takes.
+        :type kwargs: object
+        :return: A list of files
+        :rtype: list
+        """
+        return [
+            obj.name
+            for obj in self.list_directories_and_files(share_name, directory_name, **kwargs)
+            if isinstance(obj, File)
```

Comment:
What else can be returned?

Reply:
a `Directory` object

Reply:
Should we enable recursive download?

Reply:
As far as I've checked other similar operators they don't allow such thing therefore I decided to keep it that way here. IMO users might be confused if this operator acts in a different way.